README.md

```diff
@@ -0,0 +1,266 @@
+
+But wait, you say, what about the `catch` function? Good point. Well, if you noticed in the above definition, the spec says that the `then` method allows us to register *multiple* callbacks -- yes we are still using callbacks -- and one of these callbacks takes the eventual value, while the other accepts a reason the promise failed. The actual signature of the `then` function therefore is: `then(onFulfilled, onRejected)`. And the `catch` method is just a shorthand for calling `then` with only a rejection callback handler, like so: `then(null, rejectionCallback)`. The code above, in other words, could also have been written like this:
+
+```JavaScript
+makeLoan(10, 0.5)
+  .then((amountPaid) => {
+    console.log(amountPaid)
+  }).then(null, (reason) => {
+    console.log(reason);
+  });
+```
+
+Okay, so that's pretty much it! In sum: A promise is an object or function with a `then` method. The `then` method allows you to 1) know when a Promise has either been fulfilled or rejected and 2) to know either the value that resulted from its fulfilment, or the reason it failed. More generally, Promises provide us with the tools to write asynchronous code synchronously and to reason clearly about future values. Now we know, basically, what's in a Promise. So let's use them!
+
+## Code-Along: Using Promises to Database I/O
+
+In this code-along, what we'll do is play around with some async operations using a database and http requests. The scenario is that we've been tasked by a client with collecting some data about flea markets in the United States. And it just so happens that the United States Department of Agriculture has provided an open source dataset with an API that we can query to collect some of this data. So what we'll be doing is writing a simple JavaScript script to pull data from the web and insert into a Postgres database.
+
+### 1. Getting Going
+
+Let's begin by getting things set up and going through some preliminary code already provided. Your lesson directory includes a package.json file with the necessary dependencies already supplied so we can just do an `npm install` to get things started. Go ahead and do that now.
+
+Now if you open `promises-example.js`, you'll see that there's some code already present. In this code-along, we'll be using a library called [knex](http://knexjs.org) in connection with another library called [pg](https://github.com/brianc/node-postgres) for our database input/output (I/O) operations. The pg library is easy: it provides an interface the Postgres database that we'll be using for this exercise. The knex library, which is the library that we'll really be using, adds an additional layer on top of pg, which further simplifies (or "abstracts") us away from the specifics of the database we are using.
```

Comment:
It looks like `promises-example.js` isn't created quite yet -- should we add that?